### PR TITLE
Add migration guide for v2

### PR DESCRIPTION
**Pull request by smaye81**
This adds a guide to the main README for migrating from Connect v1 to Connect v2.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,6 +68,168 @@ You can serve your Connect RPCs with vanilla Node.js, or use our [server plugins
...
+  -   rpc(ElizaService, ElizaService.say, impl);
+  +   rpc(ElizaService.say, impl);
+  }
+  ```
+
+- Interceptors for streaming RPCs now use appropriate stream types. In
+  v1, the server used more exact types in interceptors, for example `UnaryRequest`
+  for server-streaming rpcs while the client always used streaming variants.
+  This was unintended behavior and has been fixed in v2. Now all streaming RPCs
+  use the `StreamRequest` and `StreamResponse` types on the server as well.
```
Comment (smaye81) [inline README.md:231]: This probably could use a code example, but wasn't sure of the best way to illustrate it.

**Review by timostamm — COMMENTED** [review]
WDYT about moving this to a separate markdown file? A year down the road, we probably don't want to have the guide take real estate on the main README, but it would be good to have stable links. A separate file would solve this.

Other changes users are likely to run into:
- When creating a message to pass in a request field, the syntax changes quite a bit. It's common outside of simple examples, so we have to show an example.
- If you use a BSR dependency for custom options, generated code will reference it. This means you have to generate with `include_imports: true`. A good example is any message that uses protovalidate.
- PlainMessage - we can repeat the information from the Protobuf-ES migration guide.
- WKT methods - for example, converting a `google.protobuf.Timestamp` to an ECMAScript Date object is no longer possible via a method. We need to call out the helpers exported from `@bufbuild/protobuf/wkt`, because they are not easily discoverable. Similar for google.protobuf.Any.

Also notable:
- In general, request and responses do not need any changes when initializers / dot notation is used, but `google.protobuf.Struct` is now generated as a more convenient `JsonObject`, and proto2 fields with default values are no longer generated as optional properties.
- SSR is generally much more simple because it doesn't require any extra steps (with proto3). This is difficult to explain in all details with a diff, but we really need to explain the general changes.

**Comment by timostamm** [comment]
Just noticed: We also need to call out:
- https://github.com/connectrpc/connect-es/pull/1231
- https://github.com/connectrpc/connect-es/pull/1225

**Comment by paul-sachs** [comment]
One thing I noticed with my migration in the BSR is that the usage of helper types like `PartialMessage` and `PlainMessage` have been changed. I think we should have some guidance on how to replace those as well. Something like:

```
`PlainMessage` is no longer required since all messages are plain objects. You may need to remove the $typeName field if using one message type in place of another.

`PartialMessage` is more or less replaced with `MessageInitShape` but requires the schema type input.
```

Not exactly sure where to put those but feels like we should mention them.

**Comment by smaye81** [comment]
OK, I think I incorporated everything into this round 2 of changes. The only thing I didn't add is the bit about SSR since that isn't really a migration change. @timostamm let me know where you think it fits best (or if it makes more sense to just link to connect docs for SSR)

File: MIGRATING.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+  is now generated as a more-convenient `JsonObject`.
+- `proto2` fields with default values are no longer generated as optional
+  properties.
+- All well-known types and helpers have been moved to an export under
+  `@bufbuild/protobuf/wkt`. For example, when converting a `google.protobuf.Timestamp`
+  to an ECMAScript `Date` object, it is no longer possible via a method and must
+  be done via helpers imported from this path.
+
+For more information on all Protobuf-ES v2 changes, see the Protobuf-ES [manual](https://github.com/bufbuild/protobuf-es/blob/main/MANUAL.md)
+as well as the [Protobuf-ES v2 migration guide](https://github.com/bufbuild/protobuf-es/blob/main/MANUAL.md#update-your-code).
```
[the remark this answers is not in the document]
Comment (timostamm) [reply MIGRATING.md:173]: Noticed two more changes:

- JSON serialization options (passed to a transport or a server plugin) have been updated in Protobuf-ES v2. The option `emitDefaultValues` is renamed:
     ```diff
     const transport = createConnectTransport({
       baseUrl: "https://demo.connectrpc.com",
       jsonOptions: {
    -    emitDefaultValues: true,
    +    alwaysEmitImplicit: true,
       },
     });
     ```
- The JSON option `typeRegistry` is also renamed:
  ```diff
  import { createRegistry } from "@bufbuild/protobuf";
  import { createConnectTransport } from "@connectrpc/connect-web";
  - import { SayRequest } from "./gen/eliza_pb";
  + import { SayRequestSchema } from "./gen/eliza_pb";
  
  const transport = createConnectTransport({
    baseUrl: "https://demo.connectrpc.com",
    jsonOptions: {
  -   typeRegistry: createRegistry(SayRequest),
  -   registry: createRegistry(SayRequestSchema),
    },
  });
  ```
  Note that registries have received a major update in Protobuf-ES v2, and are much more capable and flexible now. See the [documentation for Protobuf registries](https://github.com/bufbuild/protobuf-es/blob/v2.2.2/MANUAL.md#registries).

**Review by timostamm — COMMENTED** [review]
Nice! Left a couple of suggestions below.

> the bit about SSR

I think we do need examples for how SSR changes, and guidance for replacing usage of `PlainMessage` and `PartialMessage`. But we can do that in a separate PR.

**Comment by smaye81** [comment]
@timostamm ready for round 3.

**Pull request merged**